power-adapters-data/src/main/java/com/nextfaze/poweradapters/data/CursorData.java

```diff
@@ -0,0 +1,91 @@
+package com.nextfaze.poweradapters.data;
+
+import android.database.Cursor;
+import android.support.annotation.NonNull;
+
+import java.io.Closeable;
+
+/**
+ * Simple {@link Data} implementation backed by a {@link Cursor}. The cursor can be changed or replaced at any time.
+ * The cursor will never be closed.
+ *
+ * @param <T> The type of element this data contains.
+ */
+public abstract class CursorData<T> extends Data<T> implements Closeable {
+
+    @NonNull
+    private Cursor mData;
+
+    public CursorData() {
+    }
+
+    /**
+     * @return A new element without any data from the cursor.
+     */
+    public abstract T newElement();
+
+    /**
+     * @param element The element whose data needs to be updated with the information in the cursor.
+     * @param cursor The cursor containing the information, currently at the position containing the element.
+     */
+    public abstract void populateElement(T element, Cursor cursor);
+
+    /**
+     * Replace the current cursor (if any) with the provided cursor.
+     * @param data The new cursor. If null, the data will be empty.
+     */
+    public void setData(@NonNull Cursor data) {
+        mData = data;
+        invalidate();
+    }
+
+    @Override
+    public final int size() {
+        if (mData != null) {
+            return mData.getCount();
+        } else {
+            return 0;
+        }
+    }
+
+    @NonNull
+    @Override
+    public final T get(int position, int flags) {
+        T element = newElement();
+        mData.moveToPosition(position);
+        populateElement(element, mData);
+
+        return element;
+    }
+
+    @Override
+    public final void refresh() {
```

Comment:
Rather than leave these methods unimplemented, it'd be better if the `CursorData` accepted a function that loads the new `Cursor` asynchronously, upon `refresh()`/`reload()`. That way, it publishes its loading state while the new cursor loads, and all of the methods work as expected. I'm thinking of something similar to `ArrayData`.